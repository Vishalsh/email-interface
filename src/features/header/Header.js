import React from 'react';
import PropTypes from 'prop-types';

import Button from 'components/button/Button';
import classes from './Header.module.scss';

const Header = ({ onClickHamburger }) => {
  return (
    <header className={classes.header}>
      <div className="row">
        <div className="col-xs-6">
          <div className="row middle-xs">
            <Button type="primary" onClick={onClickHamburger}>
              <i className="icon icon-hamburger"/>
            </Button>
            <input className={classes.search} placeholder="Search for something..."/>
          </div>
        </div>

        <div className="col-xs-6">
          <div className="row end-xs middle-xs">
            <i className="icon icon-email-solid"/>
            <div>
              <Button onClick={() => alert('logout')}>
                <i className="icon icon-logout"/>
              </Button>
              <span>Log out</span>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  onClickHamburger: PropTypes.func.isRequired,
};

export default Header;
