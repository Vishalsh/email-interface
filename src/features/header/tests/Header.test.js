import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../Header';

describe('Header', () => {
  const props = {
    onClickHamburger: jest.fn()
  };
  const component = shallow(<Header {...props}/>);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should trigger onClickHamburger', () => {
    const hamburger = component.find('.icon-hamburger').get(0);

    hamburger.props.onClick();

    expect(props.onClickHamburger).toHaveBeenCalled();
  });

  it('should logout the user', () => {
    component.find('.logout').props().onClick();

    expect(window.location.href).toEqual('http://localhost/');
  });
});