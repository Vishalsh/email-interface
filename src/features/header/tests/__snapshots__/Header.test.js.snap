// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component 1`] = `
<header
  className="header"
>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <div
        className="row middle-xs"
      >
        <i
          className="icon icon-hamburger hamburgerIcon"
          onClick={[MockFunction]}
        />
        <input
          className="search"
          placeholder="Search for something..."
        />
      </div>
    </div>
    <div
      className="col-xs-6"
    >
      <div
        className="row end-xs middle-xs rightIconsRow"
      >
        <div
          className="rightIcon"
        >
          <i
            className="icon icon-email-solid"
          />
          <div
            className="badge"
          >
            <Connect(InboxUnreadEmailsCount) />
          </div>
        </div>
        <div
          className="rightIcon"
        >
          <i
            className="icon icon-bell"
          />
          <div
            className="badge"
          >
            <Badge
              className="bellBadge"
              label={8}
            />
          </div>
        </div>
        <div
          className="row middle-xs logout"
          onClick={[Function]}
        >
          <i
            className="icon icon-logout logoutIcon"
          />
          <span>
            Log out
          </span>
        </div>
      </div>
    </div>
  </div>
</header>
`;
